packages/controllers/src/snaps/SnapController.ts

```diff
@@ -1364,6 +1403,38 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Handles increasing the openSessions counter on session open.
+   *
+   * @param snapId - The snap id of the snap the session was opened with.
+   */
+  onSessionOpen(snapId: SnapId) {
+    const runtime = this.getRuntimeOrDefault(snapId);
```

Comment:
Should this also start the snap if it was 0?

Reply:
Not sure that's necessary, it'll be booted up on the first request regardless. What do you think?